openmdao/core/component.py

```diff
@@ -1048,14 +1106,14 @@ def _update_dist_src_indices(self, abs_in2out, all_abs2meta, all_abs2idx, all_si
         # of the output variables
         sizes_out = all_sizes['output']
         added_src_inds = []
-        # loop over continuous inputs
+        # loop over continuous local inputs
         for iname, meta_in in abs2meta_in.items():
             if meta_in['src_indices'] is None and iname not in abs_in2prom_info:
-                i = self._var_allprocs_abs2idx[iname]
                 src = abs_in2out[iname]
                 dist_in = meta_in['distributed']
                 dist_out = all_abs2meta_out[src]['distributed']
                 if dist_in or dist_out:
+                    i = self._var_allprocs_abs2idx[iname]
```

Comment:
is "i" even used?

Reply:
Yeah, it's on lines 1131-1132 which didn't get expanded and shown in the diff.